pkg/trace/api/profiles.go

```diff
@@ -22,48 +25,74 @@ const (
+	endpoints = append(endpoints, &traceconfig.Endpoint{Host: main, APIKey: apiKey})
+
+	if opt := "apm_config.profiling_additional_endpoints"; config.Datadog.IsSet(opt) {
+		extra := config.Datadog.GetStringMapStringSlice(opt)
+		for endpoint, keys := range extra {
+			for _, key := range keys {
+				endpoints = append(endpoints, &traceconfig.Endpoint{Host: endpoint, APIKey: key})
+			}
+		}
 	}
-	return profilingURLDefault
+	return endpoints
 }
 
-// profileProxyHandler returns a new HTTP handler which will proxy requests to the profiling intake.
-// If the URL can not be computed because of a malformed 'site' config, the returned handler will always
+// profileProxyHandler returns a new HTTP handler which will proxy requests to the profiling intakes.
+// If the main intake URL can not be computed because of config, the returned handler will always
 // return http.StatusInternalServerError along with a clarification.
 func (r *HTTPReceiver) profileProxyHandler() http.Handler {
-	target := profilingEndpoint()
-	u, err := url.Parse(target)
-	if err != nil {
-		log.Errorf("Profile forwarder is OFF because of invalid intake URL: %v", err)
-		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-			msg := fmt.Sprintf("Agent is misconfigured with an invalid intake URL: %q", target)
-			http.Error(w, msg, http.StatusInternalServerError)
-		})
+	endpoints := profilingEndpoints(r.conf.APIKey())
+	var (
+		targets []*url.URL
+		keys    []string
+	)
+	for i, endpoint := range endpoints {
+		target, err := url.Parse(endpoint.Host)
+		if err != nil {
+			log.Errorf("Error parsing profiling intake URL %s: %v", endpoint.Host, err)
+			if i == 0 {
+				// when the main intake is incorrect we don't use additional
+				return errorHandler(endpoint.Host)
+			}
+			continue
+		}
+		targets = append(targets, target)
+		keys = append(keys, endpoint.APIKey)
 	}
 	tags := fmt.Sprintf("host:%s,default_env:%s", r.conf.Hostname, r.conf.DefaultEnv)
-	return newProfileProxy(r.conf.NewHTTPTransport(), u, r.conf.APIKey(), tags)
+	return newProfileProxy(r.conf.NewHTTPTransport(), targets, keys, tags)
+}
+
+func errorHandler(endpoint string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		msg := fmt.Sprintf("Profile forwarder is OFF because of invalid intake URL configuration: %v", endpoint)
+		http.Error(w, msg, http.StatusInternalServerError)
+	})
 }
 
 // newProfileProxy creates a single-host reverse proxy with the given target, attaching
 // the specified apiKey.
-func newProfileProxy(transport http.RoundTripper, target *url.URL, apiKey, tags string) *httputil.ReverseProxy {
+func newProfileProxy(transport http.RoundTripper, targets []*url.URL, keys []string, tags string) *httputil.ReverseProxy {
 	director := func(req *http.Request) {
-		req.URL = target
-		req.Host = target.Host
-		req.Header.Set("DD-API-KEY", apiKey)
+		// URL, Host and key are set in the transport for each outbound request
```

Comment:
Can we put this comment as the very last line in the function? I feel that usually the first comment in a function (or any comment that has code below it) usually describes what is below. I am thinking that if we put it last it would be more clear that it refers to something that is missing. I also imagine a reader reaching the end of the function and only then wondering where the URL is set.